### PR TITLE
feat(appsync): Move appsync permission grants to base class

### LINKED ISSUE #23031: appsync-alpha: Add grant methods to IGraphqlApi
### Describe the feature

Grant methods are available in GraphqlApi but not in IGraphqlApi. So when you import a GraphQL Api you don't have the option to add grants. 

### Use Case

I have a shared GraphQL Api in one stack that multiple stacks use. So I would like to import it and being able to use grant methods.

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.50.0

### Environment details (OS name and version, etc.)

macOs

### PR DESCRIPTION
**Pull request by TimQuelch**
This allows the grant functions to be called for an API imported with `GraphqlApi.fromGraphqlApiAttributes`

The grant methods as they currently exist don't use any properties that are unique to the full GraphqlApi construct. They are able to be moved to GraphqlApiBase and IGraphqlApi without any behavior change.

Closes #23031 .

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by TimQuelch** [comment]
I don't think anything necessarily needs to be added to the current README, as this change should just make a imported API act more like a constructed one. Although I'm happy to modify it if anyone disagrees

Exemption Request

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks for the contribution, the implementation looks good.

A couple of additions are needed (as suggested by the linter):
1. An [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md#how-to-write-integration-tests) ([here](https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk-testing/framework-integ/test/aws-appsync/test)) to verify that granting permissions on an imported API works for better coverage
2. Some documentation about the fact that permissions are grantable on imported APIs as well (the [Permissions](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-appsync/README.md#permissions) section is a good place for it)

File: packages/aws-cdk-lib/aws-appsync/README.md [hunk 1]
```diff
@@ -519,16 +519,18 @@ the ability to add data sources and resolvers through a `IGraphqlApi` interface.
...
 });
 importedApi.addDynamoDbDataSource('TableDataSource', table);
+importedApi.grantMutation(role, 'aMutation')
 ```
 
 If you don't specify `graphqlArn` in `fromXxxAttributes`, CDK will autogenerate
-the expected `arn` for the imported api, given the `apiId`. For creating data
-sources and resolvers, an `apiId` is sufficient.
+the expected `arn` for the imported api, given the `apiId`. sources, creating
+resolvers, and granting IAM permissions, an `apiId` is sufficient.
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-appsync/README.md:533]: Let's instead add a section [here](https://github.com/aws/aws-cdk/blob/de0a9e218eeccbe4f685e45f93e25a250733dc51/packages/aws-cdk-lib/aws-appsync/README.md?plain=1#L670):
```
### Permissions on imported APIs

You can use the `grant` functions on imported GraphQL APIs as well:

```ts
declare const api: appsync.GraphqlApi;
declare const role: iam.Role;

const importedApi = appsync.GraphqlApi.fromGraphqlApiAttributes(this, 'IApi', {
  graphqlApiId: api.apiId,
  graphqlApiArn: api.arn,
});

// For generic types
importedApi.grantMutation(role, 'updateImportedExample');

// For custom types and granular design
importedApi.grant(role, appsync.IamResource.ofType('Mutation', 'updateImportedExample'), 'appsync:GraphQL');
```
Feel free to improve it further.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks for the implementation 👍
Some re-organization is needed in my opinion.

**Comment by kaizencc** [comment]
@TimQuelch are you still available to work on this? Looks like there are some outstanding action items from the latest review.

**Comment by paulhcsun** [comment]
Closing for staleness, @TimQuelch feel free to open a new PR if you'd like to continue working on this.

**Pull request closed without merging**